website/docs/developer/research/research-orca-delete-functionality.md

```diff
@@ -0,0 +1,122 @@
+
+
+- As a DAAC, I would like to configure the amount of time data remains in ORCA after being signaled to be deleted.	
+
+    This should be configurable by each DAAC that is using ORCA. Even after the data is deleted from Cumulus, it should stay in ORCA for x number of days for safety reasons.
+
+
+- As a DAAC, I would like only authorized personnel to be able to remove data from ORCA in a controlled way to minimize risk of accidental or malicious deletions.
+
+    This could be implemented with the use of NASA launchpad authentication. This [link](https://idmax.nasa.gov/nams/asset/255115) is used to submit a new GSFC workflow request and most probably will be used to only allow specific users to perform the deletes. Contact `robert.a.meredith@nasa.gov` and `tadd.buffington@nasa.gov` with any issues regarding launchpad integration.
+
+- As a DAAC, I would like duplicate data to be versioned in ORCA in order to recover the proper version of the data based on the type of failure.	
+
+
+- Perform a check with CMR and/or Cumulus to verify that the granule has been deleted from those locations prior to allowing the delete to proceed in ORCA.
+
+    
+    This can be implemented in the existing lambda function that will be used to delete the data from ORCA.
+
+### API gateway
+
+API gateway is currently used by end users for creating reconciliation reports and getting status update of recovery granules and jobs. This can be used for creating API calls made by end users for deletion. The API gateway should act as a trigger to the lambda functions that actually perform the deletion in the backend. An API gateway with two different resources need to be created- one for the soft delete API call and one for the hard delete API call.
+Check API gateway [examples](https://github.com/nasa/cumulus-orca/tree/develop/modules/api-gateway) currently used by ORCA. 
+
+### Lambda function
+
+The delete functionality can be performed by two lambda functions based on initial research. One of the lambda functions will be used to perform soft delete- delete from S3 bucket but only flag the granule in the database for deletion after 90 days. The lambda should also verify with CMR and/or Cumulus that the granule has been deleted from those locations prior to allowing the delete to proceed in ORCA. The lambda will be triggered by an API call made by the user once the user is authenticated with Launchpad.
+
+The other lambda function can be used to perform a physical delete- delete from S3 glacier as well as from the database. Similar to soft delete, it should first verify with CMR and/or Cumulus that the granule has been deleted from those locations before deleting from ORCA. This will require further discussion with Cumulus team before performing the actual work.
```

Comment:
Since you are limited to 1000 deletes per call, how would you batch this up? How would you envision this being called end user or cron? Are there better ways to do batch deletes? what are the rules this has to follow, note the use cases this gets tied to.

You may also want to describe what happens in the ORCA catalog and auditing for both soft and physical deletes. What information should we capture? Do we delete the records or move them to an audit table? How long do we keep them?

Reply:
not sure about this. I guess we will need one more research card to look into this